data/transactions/logic/assembler.go

```diff
@@ -1698,39 +1704,59 @@ func (ops *OpStream) cycle(macro string, previous ...string) bool {
 	return false
 }
 
-// Once a version is obtained, versionedMacroFilter goes through previously defined macros
+// Once a version is obtained, versionedCheckMacroNames goes through previously defined macros
 // and ensures they don't use opcodes/fields from the obtained version
-func (ops *OpStream) versionedMacroFilter() error {
-	var err error = nil
+func (ops *OpStream) versionedCheckMacroNames() error {
+	var errored bool
 	for macroName := range ops.macros {
 		if fieldNames[ops.Version][macroName] {
-			err = fmt.Errorf("%s is defined as a macro but is a field name in version %d", macroName, ops.Version)
+			ops.errorf("%s is defined as a macro but is a field name in version %d", macroName, ops.Version)
+			errored = true
+			delete(ops.macros, macroName)
 		} else if _, ok := OpsByName[ops.Version][macroName]; ok {
-			err = fmt.Errorf("%s is defined as a macro but is an opcode in version %d", macroName, ops.Version)
-		} else {
-			continue
+			ops.errorf("%s is defined as a macro but is an opcode in version %d", macroName, ops.Version)
+			errored = true
+			delete(ops.macros, macroName)
 		}
-		ops.error(err)
-		delete(ops.macros, macroName)
 	}
-	if err != nil {
-		err = errors.New("version is incompatible with defined macros")
+	if errored {
+		return errors.New("version is incompatible with defined macros")
 	}
-	return err
+	return nil
 }
 
-func (ops *OpStream) filterMacro(macroName string) error {
+func checkMacroName(macroName string, version uint64) error {
+	otherAllowedChars := [256]bool{'+': true, '-': true, '*': true, '/': true, '^': true, '%': true, '&': true, '|': true, '#': true, '!': true, '>': true, '<': true, ':': true, '=': true, '?': true}
+	for i := 0; i < len(macroName); i++ {
+		c := macroName[i]
+		if !unicode.IsLetter(rune(c)) && !unicode.IsDigit(rune(c)) && !otherAllowedChars[c] {
+			return fmt.Errorf("%s character not allowed in macro name", string(c))
```

Comment:
This is perhaps inconsequential, but the current loops iterates over bytes in the string, while this would iterate over Unicode runes:
```suggestion
	otherAllowedChars := map[rune]bool{'+': true, '-': true, '*': true, '/': true, '^': true, '%': true, '&': true, '|': true, '#': true, '!': true, '>': true, '<': true, ':': true, '=': true, '?': true}
	for c := range macroName {
		if !unicode.IsLetter(c) && !unicode.IsDigit(c) && !otherAllowedChars[c] {
			return fmt.Errorf("%s character not allowed in macro name", string(c))
```

I suggest it because you'd get a better error message when a multi-byte rune is found.

Reply:
This `c` is the index of the rune, isn't it. I think you want `for _, c := range macroName {`

https://go.dev/play/p/JOhWEHzoeFl

Reply:
My mistake, you are correct